osidb/models.py

```diff
@@ -1538,22 +1541,28 @@ def __init__(self, *args, **kwargs):
         Flaw, on_delete=models.CASCADE, related_name="snippets", blank=True, null=True
     )
 
-    def convert_snippet_to_flaw(self) -> Union[Flaw, None]:
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                name="unique_snippets", fields=["source", "external_id"]
+            ),
+        ]
+
+    def convert_nvd_snippet_to_flaw(self) -> Union[Flaw, None]:
         """
         Creates a new flaw from the snippet's content if a flaw with the given cve_id
         does not exist, and links them together. If a flaw already exists and does not
         contain the snippet of the current source, the snippet will be linked to it.
 
         Returns a flaw if it was newly created, None otherwise.
         """
-        cve_id = self.content["cve_id"]
+        cve_id = self.external_id
```

Comment:
I do not think that this change is correct. The flaw is to be created based on the Snippet content. Are we sure that it will always contain a CVE ID for any future source and that it will be the unique Snippet identifier in the source? I think we are mixing two concepts together here.

On the other hand the original code should probably use `.get` so it counts on the case when the `cve_id` is actually not present and we should just create a CVE-less flaw. Sure that this is not going to be case for NVD but this method is supposed to be collector-independent and ideally never needed to be changed.

Reply:
I overlooked this, @osoukup is right. The suggested change would work only for NVD.